src/main/java/core/basesyntax/dao/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,209 @@
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot create car " + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getByIdQuery = "SELECT * FROM cars c"
+                + " INNER JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " WHERE c.id = ?"
+                + " AND c.deleted = false AND m.deleted = false";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(getByIdQuery)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDrivers(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery = "SELECT * FROM cars c"
+                + " INNER JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " WHERE c.deleted = false AND m.deleted = false";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get any cars", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDrivers(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET manufacturer_id = ?,"
+                + " model = ? WHERE id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updatePreparedStatement = connection
+                         .prepareStatement(updateQuery)) {
+            updatePreparedStatement.setLong(1, car.getManufacturer().getId());
+            updatePreparedStatement.setString(2, car.getModel());
+            updatePreparedStatement.setLong(3, car.getId());
+            updatePreparedStatement.executeUpdate();
+            deleteFromCarsDrivers(car);
+            insertIntoCarsDrivers(car);
```

Comment:
inside these methods we open a new connection - let's move them below the try-catch block so that we do not open one connection inside another